File: TEO/Mav.UI/Mav.Ui/src/app/models/admin/update-statement.view.model.ts
import { StatementQuestionViewModel } from "./statement-question.view.model";
import { StatementAnswerViewModel } from "../statement.answer.view.model";

export class UpdateStatementViewModel {

    public id: number;

    public parentID: number;

    public statementType: number;

    public number: number;

    public content: string;

    public questions: Array<StatementQuestionViewModel>

    public answers: Array<StatementAnswerViewModel>

}

File: TEO/Mav.Models/ViewModels/UpdateStatementViewModel.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Mav.Models.ViewModels
{
    public class UpdateStatementViewModel
    {
        [Required]
        public int ID { get; set; }

        [Required]
        public int Number { get; set; }

        [Required(ErrorMessage = "Please enter a statement")]
        [Display(Name = "Title")]
        [StringLength(4000, ErrorMessage = "Statement value is too long")]
        public string Content { get; set; }

        public List<StatementQuestionViewModel> Questions { get; set; }

        public List<UserStatementAnswerViewModel> Answers { get; set; }
    }

}
